### PR TITLE
Support JSON RPC API

### PR DESCRIPTION
**Pull request by ggicci**
Support JSON RPC API

### CODE REVIEW

**Review by leventeliu — APPROVED** [review]
LGTM.

File: cmd/cqld/bootstrap.go [hunk 1]
```diff
@@ -77,6 +79,13 @@ func runNode(nodeID proto.NodeID, listenAddr string) (err error) {
 		return
 	}
 
+	if mode == "api" {
```
Comment (auxten) [inline cmd/cqld/bootstrap.go]: make "api" a const

File: cmd/cqld/main.go [hunk 2]
```diff
@@ -144,3 +153,51 @@ func main() {
 
 	log.Info("server stopped")
 }
+
+func registerNodeToBP(timeout time.Duration) (err error) {
```
Comment (auxten) [inline cmd/cqld/main.go]: why you copy this function from "cql-minerd"
if it's common usage why not to put it into some package

File: cmd/cqld/main.go [hunk 3]
```diff
@@ -76,6 +84,7 @@ func init() {
 	flag.StringVar(&memProfile, "mem-profile", "", "Path to file for memory profiling information")
 	flag.StringVar(&metricWeb, "metric-web", "", "Address and port to get internal metrics")
 
+	flag.StringVar(&wsapiAddr, "wsapi", "", "Address of the websocket JSON-RPC API")
 	flag.StringVar(&mode, "mode", "normal", "Run mode, e.g. normal, api")
```
Comment (auxten) [inline cmd/cqld/main.go]: maybe just one flag "wsapi" is enough

**Review by auxten — APPROVED** [review]
LGTM

**Pull request merged**